sprimber-cucumber/src/main/java/com/griddynamics/qa/sprimber/reporting/CucumberAllureTransformer.java

```diff
@@ -58,36 +59,33 @@ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
     private final AllureLifecycle lifecycle;
     private final Set<String> nonPrintableExceptions;
 
-    @EventListener
+    @EventListener(condition = "#root.event.node.adapterName == T(com.griddynamics.qa.sprimber.discovery.CucumberAdapterConstants).ADAPTER_NAME")
```

Comment:
How the condition is working?

Reply:
it is Spring embedded feature, to use SpEL for custom event filtering https://www.baeldung.com/spring-events#generic-listener